### PR TITLE
Do not scan uncommentable files

### PR DESCRIPTION
**Pull request by hoijui**
This prevents incidents of falsely detected copyright info
in binary files (and other uncommentable file types),
if they contain the word "Copyright" as plain-text.
This happened to me in the case of some JPG files, for example.

This also saves us from reading binary file contents unnecessarily.

... or are there ever cases where we want to detect plain-text in binary or uncommentable files?

### CODE REVIEW

**Comment by carmenbianca** [comment]
This PR seems good. I did some light tweaking. I especially liked renaming the function from `_is_uncommentable` to `_is_commentable`. What a readability improvement :sweat_smile: 

I'm not _entirely_ certain this won't have undesired effects. I'm going to approve the PR, and I'll merge it after discussing with some other peeps.

Thanks a lot @hoijui !

**Review by carmenbianca — APPROVED** [review]

**Comment by hoijui** [comment]
thank you for these comments @carmenbianca, and the improvements!
... made my day! :-)

**Comment by carmenbianca** [comment]
Did an in-person review together with @nicorikken. Let's merge this! (after fixing some merge conflicts)

**Comment by hoijui** [comment]
Wooho, thanks! :-)

**Pull request merged**
